DependencyInjection/Neo4jExtension.php

```diff
@@ -167,20 +185,23 @@ private function getUrl(array $config): string
             $config['username'],
             $config['password'],
             $config['host'],
-            $config['port']
+            $this->getPort($config)
         );
     }
 
     /**
-     * {@inheritdoc}
+     * Return the correct default port if not manually set.
+     *
+     * @param array $config
+     *
+     * @return int
      */
-    public function getConfiguration(array $config, ContainerBuilder $container): Configuration
+    private function getPort(array $config)
     {
-        return new Configuration($container->getParameter('kernel.debug'));
-    }
+        if (isset($config['port'])) {
+            return $config['port'];
+        }
 
-    public function getAlias(): string
-    {
-        return 'neo4j';
+        return 'http' == $config['schema'] ? HttpDriver::DEFAULT_HTTP_PORT : BoltDriver::DEFAULT_TCP_PORT;
```

Comment:
You should check if the BoltDriver exists. If the schema is bolt and the package does not exists the we must throw a LogicException. Right now it would be a "Class not found."